src/drivers/intel/cavs/hda-dma.c

```diff
@@ -396,16 +397,15 @@ static int hda_dma_start(struct dma *dma, int channel)
 
 static int hda_dma_release(struct dma *dma, int channel)
 {
-	/* TODO: to be removed, no longer called by anyone */
-	struct dma_pdata *p = dma_get_drvdata(dma);
+	/* Implementation left for future alignment
+	 *of dma pointers (if needed)
+	 */
 	uint32_t flags;
 
 	spin_lock_irq(&dma->lock, flags);
 
-	trace_host("Dpr");
-
-	/* resume and reload DMA */
-	p->chan[channel].status = COMP_STATE_ACTIVE;
```

Comment:
can we just remove the invoking to hda_dma_release as it actually do nothing?

Reply:
Yes we could , however after consulting this with @mmaka1 we concluded it should stay in as it will later be used for dma pointers mismatch aligment.

Reply:
That makes sense to me.
Only wondering if we will keep COMP_STATE_PAUSED for dma channels, looks nobody is using that anymore(and better to reflect this in state machine which may helps for understanding).

Reply:
Currently You are right that COMP_STATE_PAUSED is not used but  state mashine is the same for all comps. And even if we removed everything from this function we would still need empty implementation since it is dma ops.